src/core/ui/forms/MarkdownInputControls/InsertEmbedCodeButton/Iframe.ts

```diff
@@ -0,0 +1,84 @@
+import { Node } from '@tiptap/core';
+
+// We need to declare it explicitly since the iframe option is still in experimental phase.
+declare module '@tiptap/core' {
+  interface Commands<ReturnType> {
+    iframe: {
+      setIframe: (options: { src: string }) => ReturnType;
+    };
+  }
+}
+
+const defaultIframeAttributes = {
+  src: { default: null },
+  position: { default: 'absolute' },
+  top: { default: 0 },
+  left: { default: 0 },
+  width: { default: '100%' },
+  height: { default: '100%' },
+  frameborder: { default: 0 },
+  sandbox: { default: 'allow-scripts allow-same-origin allow-popups allow-presentation' },
+};
+
+export type IframeOptions = {
+  allowFullscreen: boolean;
+  HTMLAttributes: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    [key: string]: any; // It's any by default, keep it.
+  };
+};
+
+export const Iframe = Node.create<IframeOptions>({
+  name: 'iframe',
+
+  group: 'block',
+
+  atom: true,
+
+  addOptions() {
+    return {
+      allowFullscreen: true,
+      mozallowfullscreen: true,
+      webkitallowfullscreen: true,
+      HTMLAttributes: {
+        class: 'iframe-wrapper',
+        style: 'position: relative; height: 442px; width: 100%;',
```

Comment:
where does this height go?

Reply:
This `height` applies to the visually presented iframe in the active editing dialog. Without this `height` being explicitely set the iframe is short which can misslead the user that later the iframe will be visible as a short one which is not the case because when you save the dialog the iframe later is presented like it should be. Bottom line - this `height` here is for UI consistency.